ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -0,0 +1,584 @@
+           checked={!!props.value}
+           onChange={props.onChange ? (e => props.onChange(e.target.checked)) : undefined}/>
+    <div style={{display: 'flex', flexDirection: 'column', marginTop: '-0.2rem'}}>
+      <label style={styles.shortDescription}>
+        {props.item.shortDescription}
+      </label>
+      <div>
+        <label style={styles.longDescription}>
+          {props.item.longDescription}
+        </label>
+        {props.children}
+      </div>
+    </div>
+  </div>;
+};
+
+export enum WorkspaceEditMode { Create = 1, Edit = 2, Clone = 3 }
+
+
+export interface WorkspaceEditProps {
+  routeConfigData: any;
+  workspace: Workspace;
+}
+
+export interface WorkspaceEditState {
+  cdrVersionItems: Array<CdrVersion>;
+  disableButton: boolean;
+  workspace: Workspace;
+  workspaceCreationConflictError: boolean;
+  workspaceCreationError: boolean;
+  cloneUserRole: boolean;
+  loading: boolean;
+}
+
+export const WorkspaceEdit = withRouteConfigData()(withCurrentWorkspace()(
+  class extends React.Component<WorkspaceEditProps, WorkspaceEditState> {
+
+    constructor(props: WorkspaceEditProps) {
+      super(props);
+      this.isEmpty.bind(this);
+      this.state = {
+        cdrVersionItems: [],
+        disableButton: true,
+        workspace: {
+          name: '',
+          description: '',
+          dataAccessLevel: DataAccessLevel.Registered,
+          cdrVersionId: '',
+          researchPurpose: {
+            diseaseFocusedResearch: false,
+            methodsDevelopment: false,
+            controlSet: false,
+            aggregateAnalysis: false,
+            ancestry: false,
+            commercialPurpose: false,
+            population: false,
+            reviewRequested: false,
+            containsUnderservedPopulation: false,
+            underservedPopulationDetails: []
+          }
+        },
+        workspaceCreationConflictError: false,
+        workspaceCreationError: false,
+        cloneUserRole: false,
+        loading: false
+      };
+    }
+
+    componentWillMount() {
+      this.updateWorkspace();
+      this.setCdrVersions();
+    }
+
+
+    updateWorkspace() {
+      if (this.isMode(WorkspaceEditMode.Create)) {
+        const profile =  userProfileStore.getValue().profile;
+        this.setState(fp.set(['workspace', 'namespace'], profile.freeTierBillingProjectName));
+        return;
+      }
+      this.setState({workspace : this.props.workspace});
+      if (this.isMode(WorkspaceEditMode.Clone)) {
+        this.setState(
+          fp.set(['workspace', 'name'], ('Duplicate of ' + this.props.workspace.name)));
+      }
+    }
+
+    async setCdrVersions() {
+      const cdrVersions = await cdrVersionsApi().getCdrVersions();
+      // Convert cdrVersion to select input format of label and value
+      const versions = [];
+      cdrVersions.items.map((version , i) => {
+        versions.push({label: version.name, value: version.cdrVersionId});
```

Comment:
Rather than building this here, we could just store the raw data and then compute the options list during render.